api/agent/agent.go

```diff
@@ -374,6 +375,8 @@ func (a *agent) waitHot(ctx context.Context, call *call) (Slot, error) {
 			return nil, ctx.Err()
 		case <-time.After(time.Duration(200) * time.Millisecond):
 			// ping dequeuer again
+		case <-a.shutdown: // server shutdown
+			return nil, errors.New("agent shut down")
```

Comment:
since we're waiting on a slot and haven't run the call, we can return some kind of timeout error here so that fnlb might try it elsewhere (fnlb does not do this currently, but may eventually) -- would make updating an fn cluster cause less errors (slightly, in this case)